server/src/test/java/org/elasticsearch/index/search/geo/GeoPointParsingTests.java

```diff
@@ -57,6 +57,7 @@ public void testGeoPointReset() throws IOException {
         assertPointsEqual(point.reset(0, 0), point2.reset(0, 0));
         assertPointsEqual(point.resetFromString(Double.toString(lat) + ", " + Double.toHexString(lon)), point2.reset(lat, lon));
         assertPointsEqual(point.reset(0, 0), point2.reset(0, 0));
+        assertPointsEqual(point.resetFromString("POINT(" + lon + " " + lat + ")"), point2.reset(lat, lon));
```

Comment:
Could you also add a negative test that makes sure that we handling something like `"NOT A POINT"` or `"MULTIPOINT (10 40, 40 30, 20 20, 30 10)"` in a reasonable way?